File: Problem 008/solution.go
package main

import "fmt"

func max(val1, val2 int) int {
    if val1 > val2 {
        return  val1
    } else {
        return val2
    }
}

func prod(number string) int {
    prod := 1
    for _, digit := range number {
        prod *= int(digit - '0')
    }

    return prod
}

func main() {
    var cases int
    fmt.Scan(&cases)

    for c := 0; c < cases; c++ {
        var n, k, max_prod int
        fmt.Scan(&n, &k)

        var number string
        fmt.Scan(&number)

        max_prod = -1
        for i := 0; i <= n - k; i++ {
            val := prod(number[i:i + k])
            max_prod = max(val, max_prod)
        }

        fmt.Println(max_prod)
    }
}

File: Problem 008/solution.c
#include <stdio.h>

int max(int num1, int num2) {
    if (num1 > num2)
        return num1;
    else
        return num2;
}

int main(int argc, char const *argv[])
{
    int cases;
    scanf("%d", &cases);
    for (int c = 0; c < cases; ++c)
    {
        int N, K;
        long int max_prod = -1;
        scanf("%d %d", &N, &K);

        char number[N];
        scanf("%s", number);

        for (int i = 0; i < N - 4; ++i)
        {
            long int prod = 1;
            for (int j = i; j < K + i; ++j)
            {
                prod *= number[j] - '0';
            }
            max_prod = max(prod, max_prod);
        }

        printf("%ld\n", max_prod);
    }
    return 0;
}
